src/aten/sycl/PowKernels.cpp

```diff
@@ -0,0 +1,277 @@
+  auto exp = exp_scalar.to<opmath_t>();
+  auto f = PowChalfTensorScalarFunctor<scalar_t, opmath_t>(exp);
+  gpu_kernel(iter, f);
+}
+
+void pow_tensor_tensor_kernel(TensorIteratorBase& iter) {
+  auto common_dtype = iter.common_dtype();
+  if (common_dtype == kComplexHalf) {
+    using scalar_t = c10::complex<at::Half>;
+    if (iter.is_cpu_scalar(1)) {
+      const auto base = iter.scalar_value<scalar_t>(1);
+      iter.remove_operand(1);
+      pow_scalar_tensor_impl(iter, base);
+    } else if (iter.is_cpu_scalar(2)) {
+      const auto exp = iter.scalar_value<scalar_t>(2);
+      iter.remove_operand(2);
+      pow_chalf_tensor_scalar_impl(iter, exp);
+    } else {
+      using opmath_t = at::opmath_type<scalar_t>;
+      TORCH_INTERNAL_ASSERT(!iter.is_cpu_scalar(1) && !iter.is_cpu_scalar(2));
+      auto f = PowTensorTensorFunctor<opmath_t>();
+      gpu_kernel(iter, f);
+    }
+  } else {
+    AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND2(
+        kHalf, kBFloat16, iter.common_dtype(), "pow_xpu", [&] {
+          if (iter.is_cpu_scalar(1)) {
+            const auto base = iter.scalar_value<scalar_t>(1);
+            iter.remove_operand(1);
+            pow_scalar_tensor_impl(iter, base);
+          } else if (iter.is_cpu_scalar(2)) {
+            const auto exp = iter.scalar_value<scalar_t>(2);
+            iter.remove_operand(2);
+            pow_tensor_scalar_kernel(iter, exp);
+          } else {
+            gpu_kernel(iter, PowTensorTensorFunctor<scalar_t>());
+          }
+        });
+  }
+}
+
+template <typename scalar_t>
+struct PowImplUnaryFunctor1 {
+  scalar_t operator()(scalar_t base) const {
+    return base * base;
+  }
+};
+
+template <typename scalar_t>
+struct PowImplUnaryFunctor2 {
+  scalar_t operator()(scalar_t base) const {
+    return base * base * base;
+  }
+};
+
+template <typename scalar_t>
+struct PowImplUnaryFunctor3 {
+  scalar_t operator()(scalar_t base) const {
+    return 1.0 / (base * base);
+  }
+};
+
+template <typename scalar_t>
+struct PowImplUnaryFunctor4 {
+  scalar_t operator()(scalar_t base) const {
+    return 1.0 / (base * base);
+  }
+};
+
+template <typename scalar_t>
+struct PowScalarTensorFunctor2 {
+  scalar_t operator()(scalar_t base) const {
+    return pow_impl<scalar_t>(base, exp_);
+  }
+  PowScalarTensorFunctor2(scalar_t exp) : exp_(exp) {}
+
+ private:
+  scalar_t exp_;
+};
+
+template <typename Base_type, typename Exp_type>
+void pow_tensor_scalar_kernel_impl(TensorIteratorBase& iter, Exp_type exp) {
+  const auto d_exp = static_cast<double>(exp);
+  // .5 (sqrt), -.5 (rsqrt) and -1 (reciprocal) specializations are handled
+  // in pow_tensor_scalar_kernel
+  if (d_exp == 2) {
+    gpu_kernel(iter, PowImplUnaryFunctor1<Base_type>());
+  } else if (d_exp == 3) {
+    gpu_kernel(iter, PowImplUnaryFunctor2<Base_type>());
+  } else if (d_exp == -2) {
+    gpu_kernel(iter, PowImplUnaryFunctor3<Base_type>());
+  } else {
+    auto f = PowScalarTensorFunctor2<Base_type>(exp);
```

Comment:
CUDA is using `pow_` for all default `exp` and `dtype` path? We need align with CUDA.